### PR TITLE
Calculate q without auxvariables and auxkernels

### PR DESCRIPTION
**Pull request by liuusu**
Calculate q on the fly without auxvariables and auxkernels. Option is given for users to output q. Tests are added.

Ref #9559

### CODE REVIEW

**Review by jwpeterson — CHANGES_REQUESTED** [review]
@liuusu This PR can't be merged because you have inadvertently checked in a libmesh submodule update with your branch (moosebuild is trying to warn you about this).

Since you only have 1 commit, it should be pretty easy for you to fix, let us know if you require assistance.

**Comment by liuusu** [comment]
@bwspenc For testing, I did "mpiexec -n 2 moose_test-opt --n-threads=1 -i time-adaptive.i --distributed-mesh --error --no-gdb-backtrace" on my local computer and it worked fine.  The input file "time-adaptive.i" does not seem to relate to my changes.

File: modules/solid_mechanics/src/postprocessors/InteractionIntegral.C [hunk 1]
```diff
@@ -107,8 +105,37 @@ InteractionIntegral::getValue()
 Real
 InteractionIntegral::computeQpIntegral()
 {
+  return 0;
+}
+
+Real
+InteractionIntegral::computeQpIntegralPhi(const std::vector<std::vector<Real>> & phi,
```
Comment (bwspenc) [inline modules/solid_mechanics/src/postprocessors/InteractionIntegral.C]: Per our discussion, let's do this all in `computeQpIntegral()`

**Review by bwspenc — CHANGES_REQUESTED** [review]
Just a few issues as noted

**Comment by bwspenc** [comment]
@jwpeterson He fixed the libmesh submodule issue you requested changes on.

**Review by jwpeterson — DISMISSED** [review]

**Review by bwspenc — APPROVED** [review]

**Comment by bwspenc** [comment]
@liuusu You're going to need to clang-format the code.

File: modules/solid_mechanics/include/postprocessors/InteractionIntegral.h [hunk 2]
```diff
@@ -14,6 +14,12 @@
 // Forward Declarations
 class InteractionIntegral;
 
+// libMesh forward declarations
+namespace libMesh
+{
+class QBase;
```
Comment (bwspenc) [inline modules/solid_mechanics/include/postprocessors/InteractionIntegral.h]: Did you determine that you needed these?

File: modules/solid_mechanics/include/postprocessors/InteractionIntegral.h [hunk 3]
```diff
@@ -54,6 +58,12 @@ class InteractionIntegral : public ElementIntegralPostprocessor
   bool _has_symmetry_plane;
   bool _t_stress;
   Real _poissons_ratio;
+  unsigned int _ring_index;
+  unsigned int _ring_first;
+  MooseEnum _q_function_type;
+  std::vector<Real> _q_elem;
```
Comment (bwspenc) [inline modules/solid_mechanics/include/postprocessors/InteractionIntegral.h]: How about calling these `_q_curr_elem`, `_phi_curr_elem`, etc, just to be a little more clear.

File: modules/solid_mechanics/src/actions/DomainIntegralAction.C [hunk 4]
```diff
@@ -229,7 +234,7 @@ DomainIntegralAction::act()
     const std::string uo_type_name("CrackFrontDefinition");
 
     InputParameters params = _factory.getValidParams(uo_type_name);
-    params.set<MultiMooseEnum>("execute_on") = "initial nonlinear";
+    params.set<MultiMooseEnum>("execute_on") = "initial timestep_end"; // was "initial nonlinear"
```
Comment (bwspenc) [inline modules/solid_mechanics/src/actions/DomainIntegralAction.C]: Just remove the comment at the end of this line.

**Review by bwspenc — CHANGES_REQUESTED** [review]
Just a couple of little things to clean up while you're doing the clang-format. Otherwise, I think it's good to go.

**Review by bwspenc — APPROVED** [review]

**Pull request merged**